### ISSUE TITLE
terraform plan results in terraform-provider-vercel_v0.10.1 plugin panic

### ISSUE THREAD

**Issue by ableeck** [vercel/terraform-provider-vercel#79]
OS: Mac OS `12.5.1 (21G83)` on ARM64
Terraform: `v1.1.4 on darwin_arm64`

Repro:
* Copy attached `repro.tf.txt` into a folder, and rename to `repro.tf`
* Change the `git_repository` config to something useful
* `terraform init`
* Set an API token with access to your Vercel team
* Have your Team ID ready for pasting
* `terraform plan`

Observed:
Crash

Output:
```
Stack trace from the terraform-provider-vercel_v0.10.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x1055442a0]

goroutine 31 [running]:
github.com/vercel/terraform-provider-vercel/client.(*Client).GetTeam(0x0, {0x1057f63e0, 0x14000282e40}, {0x1400019e420, 0x1d})
	github.com/vercel/terraform-provider-vercel/client/team_get.go:13 +0x50
github.com/vercel/terraform-provider-vercel/vercel.(*vercelProvider).Configure(0x14000120878, {0x1057f63e0, 0x14000282e40}, {{0x140000383f8, 0x5}, {{{0x1057fa008, 0x14000283410}, {0x1057206e0, 0x140002831a0}}, {0x1400049b470, ...}}}, ...)
	github.com/vercel/terraform-provider-vercel/vercel/provider.go:130 +0x274
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0x1400017bce0, {0x1057f63e0, 0x14000282e40}, 0x140002a7400, 0x14000046ac0)
	github.com/hashicorp/terraform-plugin-framework@v0.13.0/internal/fwserver/server_configureprovider.go:15 +0xc4
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0x1400017bce0, {0x1057f63e0?, 0x14000282cf0?}, 0x5?)
	github.com/hashicorp/terraform-plugin-framework@v0.13.0/internal/proto6server/server_configureprovider.go:36 +0x220
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0x14000392500, {0x1057f63e0?, 0x140002823c0?}, 0x14000046880)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov6/tf6server/server.go:527 +0x250
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0x1057c9280?, 0x14000392500}, {0x1057f63e0, 0x140002823c0}, 0x140000a4460, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140003061e0, {0x1057fa2f8, 0x1400029c4e0}, 0x14000000c60, 0x140003d7d10, 0x105c1ab98, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1301 +0x9c4
google.golang.org/grpc.(*Server).handleStream(0x140003061e0, {0x1057fa2f8, 0x1400029c4e0}, 0x14000000c60, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1642 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.49.0/server.go:938 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.49.0/server.go:936 +0x290

Error: The terraform-provider-vercel_v0.10.1 plugin crashed!
```

[repro.tf.txt](https://github.com/vercel/terraform-provider-vercel/files/9687985/repro.tf.txt)

**Comment by dglsparsons** [comment]
Hi @ableeck, thanks for raising this - it looks like an issue on our side. I'll investigate and get back to you ASAP. 

In the meantime, I believe removing the `team` setting from the `provider` config will prevent this issue.

**Comment by dglsparsons** [comment]
Hi @ableeck  - this should be resolved in the [v0.10.2 release](https://github.com/vercel/terraform-provider-vercel/releases/tag/v0.10.2). Please give it a go and let me know if the issue persists! 

Side note: 
In the configuration you provided, the `team_id` parameter is specified a lot and is redundant for most of the cases - if specified in a `provider` block, you don't need to pass it to each `resource` unless you would like the `resource` to have a different `team_id`. 

For example, this would be equivalent. 
```terraform 
terraform {
  required_providers {
    vercel = {
      source  = "vercel/vercel"
      version = "0.10.1"
    }
  }
  required_version = "~> 1.1.4"
}

variable "vercel_team_id" {
  type        = string
  description = "Vercel Team ID"
  sensitive   = true
}

provider "vercel" {
  team = var.vercel_team_id
}

resource "vercel_project" "repro_vercel_project" {
  build_command = "nuxt generate"
  framework     = "nuxtjs"
  git_repository = {
    repo = "{redacted}"
    type = "github"
  }
  install_command            = "npm ci"
  name                       = "repro"
  serverless_function_region = "iad1"
}

resource "vercel_project_domain" "repro_vercel_project_domain_production" {
  domain     = "repro.example.com"
  project_id = vercel_project.repro_vercel_project.id
}

resource "vercel_project_domain" "repro_vercel_project_domain_stage" {
  git_branch = "staging"
  domain     = "stage-repro.example.dev"
  project_id = vercel_project.repro_vercel_project.id
}

resource "vercel_project_environment_variable" "repro_vercel_project_env_preview_external_login_route" {
  key        = "EXTERNAL_LOGIN_ROUTE"
  project_id = vercel_project.repro_vercel_project.id
  target = [
    "preview",
  ]
  value   = "https://login.example.dev"
}

resource "vercel_project_environment_variable" "repro_vercel_project_env_stage_external_login_route" {
  git_branch = "staging"
  key        = "EXTERNAL_LOGIN_ROUTE"
  project_id = vercel_project.repro_vercel_project.id
  target = [
    "preview",
  ]
  value   = ""
}
```

**Comment by ableeck** [comment]
Thank you, @dglsparsons!

* The fix in `0.10.2` works for our configuration
* Also thanks for the comment on the `team_id`, should be easier to read without the repetition!
